Importing necessary libraries and dataset





In [4]:

import pandas as pd


df = pd.read_csv(r"C:\Users\vansh\Downloads\teachers_rating_an.csv")


print("Columns in dataset:", df.columns.tolist())


rating_col = None
for col in df.columns:
    if 'score' in col.lower() or 'rating' in col.lower():
        rating_col = col
        break

if rating_col is None:
    raise ValueError("No column with 'score' or 'rating' found!")

print("Using column:", rating_col)




Columns in dataset: ['prof', 'age', 'eval', 'beauty', 'rating', 'students', 'gender', 'tenure', 'division']
Using column: rating


Q1. Using the teachers' rating dataset, what is the probability of receiving an evaluation score of greater than 4.5

In [5]:

prob_above_4_5 = len(df[df[rating_col] > 4.5]) / len(df)
print("Probability of evaluation score > 4.5:", prob_above_4_5)


Probability of evaluation score > 4.5: 0.24166666666666667


The probability of a teacher getting an evaluation score greater than 4.5 is:

P(score>4.5)=0.242≈24%

This means that if we randomly select a teacher, there is a 24% chance that the teacher will receive a rating higher than 4.5. So roughly 1 out of 4 teachers gets an excellent rating.

Q2. Using the teachers' rating dataset, what is the probability of receiving an evaluation score greater than 3.5 and less than 4.2

In [6]:

prob_between_3_5_4_2 = len(df[(df[rating_col] > 3.5) & (df[rating_col] < 4.2)]) / len(df)
print("Probability of evaluation score between 3.5 and 4.2:", prob_between_3_5_4_2)

Probability of evaluation score between 3.5 and 4.2: 0.43333333333333335


The probability of a teacher getting an evaluation score between 3.5 and 4.2 is:
P(3.5<score<4.2)=0.417≈42%
If we randomly select a teacher, there is a 42% chance that the teacher will receive a rating in the moderate range between 3.5 and 4.2. Most teachers fall in this average rating range.

Q3. Using the two-tailed test from a normal distribution:
1. A professional basketball team wants to compare its performance with that of players in a regional league.
2. The pros are known to have a historic mean of 12 points per game with a standard deviation of 5.5.
3. A group of 36 regional players recorded on average 10.7 points per game.
4. The pro coach would like to know whether his professional team scores on average are different from that of the regional players.
State the null hypothesis
Null Hypo: ("The mean point of the regional players is not different from the historic mean")
Alternative Hypo: ("The mean point of the regional players is different from the historic mean")

The null hypothesis (H₀) states that the mean points of regional players is not different from the historic mean of 12 points per game.
The alternative hypothesis (H₁) states that the mean points of regional players is different from the historic mean.

Using the given data: the historic mean μ = 12, standard deviation σ = 5.5, sample mean x̄ = 10.7, and sample size n = 36.



In [7]:
import math
import scipy.stats as stats


mu = 12       
sigma = 5.5  
x_bar = 10.7  
n = 36       


z = (x_bar - mu) / (sigma / math.sqrt(n))
print("Z-score:", z)


p_value = 2 * (1 - stats.norm.cdf(abs(z)))
print("Two-tailed p-value:", p_value)


alpha = 0.05
if p_value < alpha:
    print("Reject the null hypothesis: There is a significant difference.")
else:
    print("Fail to reject the null hypothesis: No significant difference.")


Z-score: -1.418181818181819
Two-tailed p-value: 0.15613768876786005
Fail to reject the null hypothesis: No significant difference.


The two-tailed p-value corresponding to z = -1.42 is approximately 0.156.
Since p-value (0.156) > 0.05, we fail to reject the null hypothesis. This means there is no significant difference between the mean points of regional players and the historic mean of professional players.